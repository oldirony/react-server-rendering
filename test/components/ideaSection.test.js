import React from 'react';
import IdeaSection from '../../src/components/idea-section/idea-section';
import classes from '../../src/components/idea-section/idea-section.scss';

const props = {
	title: 'Section title',
	index: 1,
	updateSection: ()=>true,
	isCompleted: false
}
test('Base rendering', ()=>{
	let ideaSection = shallow(<IdeaSection {...props} />);
	expect(ideaSection).toMatchSnapshot();

	ideaSection = shallow(<IdeaSection {...props} isCompleted={true} />);
	expect(ideaSection).toMatchSnapshot();
});

test('Complete button click', ()=>{
	const updateSection = jest.fn();
	const ideaSection = mount(<IdeaSection {...props} updateSection={updateSection} />);

	ideaSection.find(`.${classes.button}`).simulate('click');

	expect(updateSection).toBeCalled();

});