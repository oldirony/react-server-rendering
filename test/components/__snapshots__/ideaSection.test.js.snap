// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base rendering 1`] = `
<li
  className="section"
>
  <h4
    className="title"
  >
    Section title
  </h4>
  <p />
  <button
    className="vanilla withIcon button"
    onClick={[Function]}
  >
    <svg
      className="icon"
    >
      <use
        xlinkHref="#checkbox-unchecked"
      />
    </svg>
    Completed
  </button>
</li>
`;

exports[`Base rendering 2`] = `
<li
  className="section isCompleted"
>
  <h4
    className="title"
  >
    Section title
  </h4>
  <p />
  <button
    className="vanilla withIcon button"
    onClick={[Function]}
  >
    <svg
      className="icon"
    >
      <use
        xlinkHref="#checkbox-checked"
      />
    </svg>
    Completed
  </button>
</li>
`;
